File: benchmark/cases/python_en_13_ecs_event_bus/workspace/world.py
class World:
    def __init__(self, event_bus):
        self.entities = {}
        self.event_bus = event_bus
        self.next_id = 1

    def create_entity(self):
        eid = self.next_id
        self.next_id += 1
        self.entities[eid] = {}
        return eid

    def add_component(self, eid, component):
        comp_type = type(component).__name__
        self.entities[eid][comp_type] = component

File: benchmark/cases/node_en_16_ecs_data_driven/workspace/World.js
class World {
    constructor() {
        this.entities = new Map(); // ID -> Set of components
        this.nextId = 1;
    }

    createEntity() {
        const id = this.nextId++;
        this.entities.set(id, new Map());
        return id;
    }

    addComponent(entityId, component) {
        const entity = this.entities.get(entityId);
        if (entity) {
            entity.set(component.constructor.name, component);
        }
    }

    getEntitiesWith(componentNames) {
        const results = [];
        for (const [id, components] of this.entities) {
            if (componentNames.every(name => components.has(name))) {
                results.push({ id, components });
            }
        }
        return results;
    }
}

module.exports = World;
